lib/manageiq/messaging/kafka/common.rb

```diff
@@ -53,6 +53,7 @@ def queue_for_publish(options)
           body, kafka_opts = for_publish(options)
           kafka_opts[:headers][:message_type] = options[:message] if options[:message]
           kafka_opts[:headers][:class_name] = options[:class_name] if options[:class_name]
+          kafka_opts[:headers][:encoding] = options[:encoding] if options[:encoding]
```

Comment:
@Ladas will this behavior (not needing to JSON.parse() in the subscriber) be consistent if using artemis?

Reply:
ah, probably not, if I am changing this just in kafka?

Reply:
Yeah so e.g. if the calling code switches to artemis would https://github.com/ManageIQ/topological_inventory-ingress_api/pull/47/files#diff-fdb60c134b290dc6cd1033a065180563R27 break? IDK

Reply:
I think we'll need some integration test for this, there might be more issues switching to artemis :-)

so you want to add it also here?
https://github.com/ManageIQ/manageiq-messaging/blob/f93bcf299e13f7e3f89f9562c3f2908a23d854cc/lib/manageiq/messaging/stomp/common.rb#L23

Reply:
I'm wondering if kafka auto-parses the payload and artemis doesn't we might have to call JSON.parse on the artemis consumer side

Reply:
I believe the parsing is on our side in:

https://github.com/ManageIQ/manageiq-messaging/blob/7ce87bccfeab0aef3d3681b1afd8adee9ee241c1/lib/manageiq/messaging/common.rb#L21

Reply:
Ah okay, then yeah just adding encoding to the headers in stomp/common should do the trick :+1:

Reply:
done